### PR TITLE
[ML] Increase wait for forecasts in FIPS JVM

### LINKED ISSUE #63793: [CI] ForecastIT.testOverflowToDisk
**Build scan**: https://gradle-enterprise.elastic.co/s/dakbqxzmvfxnm 

**Repro line**: 
```
/gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.ml.integration.ForecastIT.testOverflowToDisk" -Dtests.seed=E001230D5CCE4F9 -Dtests.security.manager=true -Dtests.locale=sr-CS -Dtests.timezone=Pacific/Easter -Druntime.java=8 -Dtests.fips.enabled=true
```
**Reproduces locally?**: No

**Applicable branches**: 7.9 onwards

**Failure history**:

**Failure excerpt**:
```
java.lang.AssertionError: 
Expected: one of {<finished>}
     but: was <scheduled>
	at __randomizedtesting.SeedInfo.seed([E001230D5CCE4F9:41D0AC6ABCB2ED49]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ml.integration.MlNativeAutodetectIntegTestCase.lambda$waitForecastStatus$1(MlNativeAutodetectIntegTestCase.java:301)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:951)
	at org.elasticsearch.xpack.ml.integration.MlNativeAutodetectIntegTestCase.waitForecastStatus(MlNativeAutodetectIntegTestCase.java:298)
	at org.elasticsearch.xpack.ml.integration.MlNativeAutodetectIntegTestCase.waitForecastToFinish(MlNativeAutodetectIntegTestCase.java:292)
	at org.elasticsearch.xpack.ml.integration.ForecastIT.testOverflowToDisk(ForecastIT.java:244)
```

### PR DESCRIPTION
**Pull request by droberts195**
Forecast tests have been seen to fail in the FIPS JVM
because they take far longer than without FIPS.

This change waits up to 5 minutes for forecasts in FIPS
mode (compared to the usual 1 minute).

Fixes #63793

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java [hunk 1]
```diff
@@ -290,17 +290,25 @@ protected String forecast(String jobId, TimeValue duration, TimeValue expiresIn,
...
     protected void waitForecastToFinish(String jobId, String forecastId) throws Exception {
-        waitForecastStatus(jobId, forecastId, ForecastRequestStats.ForecastRequestStatus.FINISHED);
+        // Forecasts can take an eternity to complete in the FIPS JVM
+        waitForecastStatus(inFipsJvm() ? 300 : 60, jobId, forecastId, ForecastRequestStats.ForecastRequestStatus.FINISHED);
     }
 
     protected void waitForecastStatus(String jobId,
                                       String forecastId,
                                       ForecastRequestStats.ForecastRequestStatus... status) throws Exception {
+        waitForecastStatus(30, jobId, forecastId, status);
```
Comment (przemekwitek) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java:300]: Why is the default here (`30`) different than it is above (`60`)?

Comment (droberts195) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java:300]: The wait was 30 everywhere prior to #64471 (which relates to the same test failure in the forecast overflow to disk tests).  It was increased to 60 in that PR, but the only place a higher value was ever required was when waiting for a forecast to finish, not for waits on other statuses.  So now there is the opportunity to have a different length of wait for finish I put the length of wait for other statuses back to the original level.

Comment (przemekwitek) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java:300]: Great, thanks for explanation!

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**